### PR TITLE
docs: go sdk

### LINKED ISSUE #314: Go SDK document
- [ ] How to install
- [ ] Write data
- [ ] Query data
- [ ] API Doc

### PR DESCRIPTION
**Pull request by yuanbohan**
resolve #314

### CODE REVIEW

File: docs/reference/sdk/go.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Go SDK
+
+A Go Client for GreptimeDB, which is compatible with GreptimeDB protocol and lightweight.
+
+## Features
+
+- concurrent safe
```
Comment (sunng87) [inline docs/reference/sdk/go.md:7]: Capital case

**Comment by yuanbohan** [comment]
Sorry, but I did not find how to show `Go SDK` in preference tab.

**Comment by sunng87** [comment]
See `summary.yml`

File: docs/user-guide/write-data.md [hunk 2]
```diff
@@ -60,8 +60,44 @@ CompletableFuture<Result<WriteOk, Err>> write(WriteRows rows, Context ctx);
...
-TODO -->
+### Go
+
+To begin with, we have to prepare a `Series`, which delegates one row data. There are three kind fields in `Series` you can use:
+
+| Kind      | Description                                                                         |
+|-----------|-------------------------------------------------------------------------------------|
+| Tag       | index column, helps to retrieve data more efficiently                               |
+| Field     | value column, helps to analysis, aggregation, calculating, etc,.                    |
+| Timestamp | timestamp column, each table MUST have exactly one timestamp column, with precision |
```
Comment (nicecui) [inline docs/user-guide/write-data.md:71]: Are both seconds and millionseconds ok?

**Review by nicecui — COMMENTED** [review]
👍

File: docs/reference/sdk/go.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+- Non-blocking
+- Easy to use with Metric and Series struct
+- Support Sql and PromQL in querying
+
+## How to use
+
+Please refer to User Guide chapter to learn [how to insall SDK](/user-guide/clients.md#go-sdk),
+[write data](/user-guide/write-data.md#go) and [query data](/user-guide/query-data.md#go).
+
+## Config
```
Comment (nicecui) [inline docs/reference/sdk/go.md:17]:
```suggestion
## Configuration
```

File: docs/reference/sdk/go.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+| Name     | Description                                            |
+|:---------|:-------------------------------------------------------|
+| Addr     | GreptimeDB server address.                             |
+| Port     | default is 4001                                        |
+| Username | for GreptimeCloud. Leave it empty in local environment |
+| Password | for GreptimeCloud. Leave it empty in local environment |
```
[the remark this answers is not in the document]
Comment (yuanbohan) [reply docs/reference/sdk/go.md:24]: That's a good point

**Review by nicecui — APPROVED** [review]
Great 👍

**Review by sunng87 — APPROVED** [review]

File: docs/reference/sdk/go.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+## How to use
+
+Please refer to User Guide chapter to learn [how to insall SDK](/user-guide/clients.md#go-sdk),
+[write data](/user-guide/write-data.md#go) and [query data](/user-guide/query-data.md#go).
+
+## Configuration
+
+| Name     | Description                                                           |
+|:---------|:----------------------------------------------------------------------|
+| Addr     | GreptimeDB server address.                                            |
```
Comment (xtang) [inline docs/reference/sdk/go.md:21]:
```suggestion
| Addr     | GreptimeDB server address                                            |
```

File: docs/user-guide/query-data.md [hunk 6]
```diff
@@ -168,9 +168,49 @@ The code will output the details of querying which will look something like this
 CompletableFuture<Result<QueryOk, Err>> query(QueryRequest req, Context ctx);
 ```
 
-<!-- ### Go
-TODO -->
+### Go
 
+To begin with, you have to prepare a `QueryRequest`, and you can retrieve data from GreptimeDB via:
+
+- Sql
```
Comment (xtang) [inline docs/user-guide/query-data.md:175]:
```suggestion
- SQL
```

File: docs/user-guide/write-data.md [hunk 7]
```diff
@@ -60,8 +60,57 @@ CompletableFuture<Result<WriteOk, Err>> write(WriteRows rows, Context ctx);
...
+
+To begin with, we have to prepare a `Series`, which delegates one row data. There are three kind fields in `Series` you can use:
+
+| Kind      | Description                                                         |
+|-----------|---------------------------------------------------------------------|
+| Tag       | index column, helps to retrieve data more efficiently               |
+| Field     | value column, helps to analysis, aggregation, calculating, etc,.    |
+| Timestamp | timestamp column, each table MUST have exactly one timestamp column |
+
+then, you can add one `Series` into `Metric`, then create a `InsertRequest`, call `client.Insert` to store the data into GreptimeDB.
```
Comment (xtang) [inline docs/user-guide/write-data.md:73]:
```suggestion
then, you can add one `Series` into `Metric`, then create an `InsertRequest` and call `client.Insert` to store the data into GreptimeDB.
```

**Comment by xtang** [comment]
Almost LGTM.

**Review by xtang — APPROVED** [review]
LGTM

**Pull request merged**